### PR TITLE
run-spec-wasm2c.py: split `run_spec_tests` function if it's too long

### PR DESCRIPTION
**Pull request by keithw**
This is intended to improve GCC's long compile times for some of the extremely long SIMD tests that will come soon (#2021).

On my laptop running Ubuntu 22.04, this change reduces the time for the simd_f32x4_pmin_pmax.wast test from 16 minutes to about 70 seconds.

### CODE REVIEW

File: test/run-spec-wasm2c.py [hunk 1]
```diff
@@ -147,7 +157,7 @@ def GetModulePrefixUnmangled(self, idx):
         return self.unmangled_names[idx]
 
     def GetModuleInstanceName(self, idx_or_name=None):
-        return self.GetModulePrefix() + '_instance'
+        return self.GetModulePrefix(idx_or_name) + '_instance'
```
Comment (sbc100) [inline test/run-spec-wasm2c.py:160]: Is this related?

File: test/run-spec-wasm2c.py [hunk 2]
```diff
@@ -243,9 +253,15 @@ def _WriteCommand(self, command):
...
         self.out_file.write('%s_init_module();\n' % self.GetModulePrefix())
-        self.out_file.write('%s_instance_t %s;\n' % (self.GetModulePrefix(), self.GetModuleInstanceName()))
         self._WriteModuleInitCall(command, False)
 
+    def _WriteModuleInstances(self):
+        idx = 0
+        for command in self.commands:
+            if IsModuleCommand(command):
+                self.out_file.write('%s_instance_t %s;\n' % (self.GetModulePrefix(idx), self.GetModuleInstanceName(idx)))
+                idx += 1
```
Comment (sbc100) [inline test/run-spec-wasm2c.py:263]: Same here, is this related to the splitting up?

**Review by sbc100 — APPROVED** [review]

File: test/run-spec-wasm2c.py (hunk 1 above)
Comment (keithw) [reply test/run-spec-wasm2c.py:160]: Yes, otherwise it's not possible to call GetModuleInstanceName on an idx that isn't the current idx. (Which we need to do now that we're declaring all the module instance structs upfront.) Probably this method should always have been passing the `idx_or_name` argument through.

File: test/run-spec-wasm2c.py (hunk 2 above)
Comment (keithw) [reply test/run-spec-wasm2c.py:263]: Yes, this is most of the first commit. We're declaring all of the instance_t structs upfront (as globals) so that any of the split-up run_spec_tests_**N** functions can refer to any of them.

**Review by sbc100 — APPROVED** [review]
lgtm!

**Comment by keithw** [comment]
Sigh, this seems to be an asan false-positive in GCC 9 (which we use in CI) that's gone by GCC 11 and gone in clang.
